# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1320168623 14400
#      Tue Nov 01 13:30:23 2011 -0400
# Node ID 7c0fcabd1a661bdc16cc391f60d611504de44515
# Parent  b0b6aa4f7302b693739099291c4d05bb501fb8ca
__patch__: fix thinko in previous change.

* __patch__.m (setdata): Correctly index NaN values and last non-NaN
value in column.

diff --git a/scripts/plot/private/__patch__.m b/scripts/plot/private/__patch__.m
--- a/scripts/plot/private/__patch__.m
+++ b/scripts/plot/private/__patch__.m
@@ -234,19 +234,19 @@ function args = setdata (args)
   else
     fc = args {idx};
   endif
 
   nr = size (faces, 2);
   nc = size (faces, 1);
   idx = faces .';
   t1 = isnan (idx);
-  for i = find (any(t1))
-    first_idx_in_column = find (idx(:,i), 1);
-    idx(first_idx_in_column+1:end,i) = idx(first_idx_in_column,i);
+  for i = find (any (t1))
+    first_idx_in_column = find (t1(:,i), 1);
+    idx(first_idx_in_column:end,i) = idx(first_idx_in_column-1,i);
   endfor
   x = reshape (vert(:,1)(idx), size (idx));
   y = reshape (vert(:,2)(idx), size (idx));
   if (size(vert,2) > 2)
     z = reshape (vert(:,3)(idx), size (idx));
   else
     z = [];
   endif
